src/bun.js/event_loop.zig

```diff
@@ -494,7 +494,7 @@ pub fn waitForPromise(this: *EventLoop, promise: JSC.AnyPromise) void {
                 }
             }
         },
-        else => {},
+        .fulfilled, .rejected => this.drainMicrotasks() catch return,
```

Comment:
```suggestion
        .fulfilled, .rejected => this.drainMicrotasks() catch return, // TODO: properly propagate exception upwards
```

Doesn't have to block this PR but I think we should aspire to thread `JSExecutionTerminated` through more of the call stack